### ISSUE TITLE
Renaming detecting

### ISSUE THREAD

**Issue by diverger** [sublimehq/sublime_merge#1448]
When I rename one file. SM will record it as a file deletion and a new file addition. May SM detect the 'renaming' and record it as a name change? Then the file change history is retained.

**Comment by willrowe** [comment]
What happens when you stage the deletion and addition?

**Comment by diverger** [comment]
Sorry, this should be a feature demanding.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by willrowe** [comment]
The reason I asked that is that, unless you are experiencing a bug, it should detect the rename when you stage the deletion and addition.

**Comment by diverger** [comment]
No, they are showed as two files when I commit them at least. I will check it in the history, may there they are treated as renaming?  I'm a newer to SM.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by diverger** [comment]
I've checked the history, the renamed file are treat as two separate ones. The *new* file doesn't inherit the *old* file's history.

**Comment by willrowe** [comment]
I am not sure about this, but that may happen if you rename the file and change the contents of it in the same commit. Regardless though, I do not believe that is SM specific, it just follows how Git functions. If there is a discrepancy between how SM handles this and how Git handles it on the CLI, then there would be a bug.

**Comment by diverger** [comment]
Yes, usually renaming is companioned with content changing, so this may be the reason.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by srbs** [comment]
possible duplicate: #384

**Comment by andrejp88** [comment]
I ran into this issue just now. In the CLI Git identifies a rename, but Sublime Merge shows it as an addition and a deletion.

CLI:
![Screenshot_20220825_205904](https://user-images.githubusercontent.com/7419746/186757364-77b640fd-13bd-489b-9071-5425e115f502.png)

Sublime Merge:
![Screenshot_20220825_205916](https://user-images.githubusercontent.com/7419746/186757404-296d1e25-8d1c-4135-b7b9-7f96b6023051.png)

**Comment by dpjohnst** [comment]
Hi @andrejp88,

Thanks for reaching out!

The rename detection uses a similarity content-based heuristic to determine if a rename has occurred, and doesn't always match what the command line behavior. Could you confirm if there are significant content changes to the renamed file?

Kind regards,
\- Dylan

**Comment by andrejp88** [comment]
Hi @dpjohnst, indeed there were quite a few changes to the file. `git diff` reports a similarity index of 54%. At a glance, I would say most lines were unchanged, and those that did change only modified or removed one or two short segments of text. I've attached an excerpt from the diff shown by the Git CLI. To the human eye, it's obvious that it's the same file, but I suppose if the algorithm only looks at lines modified, then it looks quite different.

![Screenshot_20221019_122232](https://user-images.githubusercontent.com/7419746/196677520-016422e7-aaf4-48d0-8dd6-c6e207ce59f6.png)

**Comment by yaroslavyaroslav** [comment]
I've got the same issue . As for me this given function is not quite reliable. Here's an [example](https://github.com/rust-blockchain/evm/commit/4ecc2446354457309b712e5a482c97277c2f0230). This GitHub figures this commit as it is as renaming with, while SublimeMerge commit shows:
- `Cargo.toml` is renamed.
-  `main.rs` as new one.
